alf/nest/utils.py

```diff
@@ -86,6 +86,34 @@ def _combine_flat(self, tensors):
         return torch.cat(tensors, dim=self._dim)
 
 
+@gin.configurable
+class NestSelectiveConcat(NestCombiner):
+    def __init__(self, ind, dim=-1, name="NestSelectiveConcat"):
+        """A combiner for selecting from the tensors in a nest and then
+        concatenating them along a specified axis. It assumes that all the
+        selected tensors have the same tensor spec. Can be used as a
+        preprocessing combiner in ``EncodingNetwork``.
+
+        Args:
+            ind (int|list[ind]): the index or a list of indices indicating
```

Comment:
A NestCombiner assumes a general nested structure, not necessarily a flat list. I think your `ind` should also assumes a matched structure.

Reply:
Change and also merged with NestConcat.